### PR TITLE
Update LINUX_CI_OS_VERSION to Ubuntu 22.04

### PR DESCRIPTION
**Pull request by sjackman**
- Change `LINUX_CI_OS_VERSION` from `Ubuntu 16.04` to `Ubuntu 22.04`
- Change `LINUX_GLIBC_CI_VERSION` from `2.23` to `2.35`
- Change `LINUX_GCC_CI_VERSION` from `5.0` to `11.0`
- Change `LINUX_PREFERRED_GCC_FORMULA` from `gcc@5` to `gcc@11`
- Build the Docker image `ghcr.io/homebrew/ubuntu22.04:master`

-----

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

- https://github.com/Homebrew/brew/issues/13619

### CODE REVIEW

File: .github/workflows/tests.yml
Comment (Bo98) [inline .github/workflows/tests.yml:189]: How do we handle compatibility for those currently using `homebrew/ubuntu16.04:master`?

Update both? Alias `master` to `latest`?

Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:189]: @Bo98 what are the compatibility issues here?

**Review by MikeMcQuaid — COMMENTED** [review]
Looks good so far! Any docs that need updated?

File: .github/workflows/tests.yml
Comment (Bo98) [reply .github/workflows/tests.yml:189]: Existing workflows which reference `homebrew/ubuntu16.04:master`, which would suddenly stop being updated with this change (but `homebrew/ubuntu16.04:latest` still will).

File: Library/Homebrew/extend/os/linux/compilers.rb
Comment (carlocab) [inline Library/Homebrew/extend/os/linux/compilers.rb:9]: Can we just get rid of this and use the latest GCC?

I think this is part of our discussion here: https://github.com/Homebrew/brew/pull/13577#discussion_r946494611

Comment (Bo98) [reply Library/Homebrew/extend/os/linux/compilers.rb:9]: Note that this would mean brewed GCC would always be installed on CI as well since it would see GCC 11 as too old.  Note sure if this is what you're arguing or not - the prior discussion only mentions runtime libraries.

(If not, further modifications could however be made so host GCC 11 is used if available, but latest GCC is installed for runtime if older than 11.)

Comment (MikeMcQuaid) [reply Library/Homebrew/extend/os/linux/compilers.rb:9]: I feel like:
- ideally we'd use the system `glibc` and `gcc` on our primary CI builds
- users end up using the same `glibc` and `gcc` versions from the host or from us if their host version is newer or older

Comment (carlocab) [reply Library/Homebrew/extend/os/linux/compilers.rb:9]: > (If not, further modifications could however be made so host GCC 11 is used if available, but latest GCC is installed if older than 11.)

Yes, this is what I'm looking for. Apologies for the lack of clarity there.

Comment (MikeMcQuaid) [reply Library/Homebrew/extend/os/linux/compilers.rb:9]: > > (If not, further modifications could however be made so host GCC 11 is used if available, but latest GCC is installed if older than 11.)
> 
> Yes, this is what I'm looking for. Apologies for the lack of clarity there.

IMO we'll be in a better place if we just use the same GCC major version all the time.

Comment (sjackman) [reply Library/Homebrew/extend/os/linux/compilers.rb:9]: I agree that using the same GCC major version is simpler. See https://github.com/Homebrew/brew/pull/13577#discussion_r951632245

**Comment by MikeMcQuaid** [comment]
Wanting this to land for 3.6.0.

File: .github/workflows/tests.yml
Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:189]: @Bo98 Ah, I see. Yeh, feels like it would be nice to push both tags.

Comment (sjackman) [reply .github/workflows/tests.yml:189]: We push `homebrew/ubuntu16.04:latest` with each stable release of Homebrew. I'd like to permanently remove `homebrew/ubuntu16.04:master` after we updated to `homebrew/ubuntu22.04:master`.

Comment (Bo98) [reply .github/workflows/tests.yml:189]: The problem is communicating that to third party taps (e.g. https://github.com/brewsci/homebrew-num/blob/master/.github/workflows/upload-bottles.yml). We should ideally provide some deprecation messaging.

Ignoring `master` for a moment, what's your thoughts on the future of the Ubuntu 16.04 image in general?

Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:189]: > The problem is communicating that to third party taps (e.g. https://github.com/brewsci/homebrew-num/blob/master/.github/workflows/upload-bottles.yml). We should ideally provide some deprecation messaging.

Agreed.

**Comment by sjackman** [comment]
https://github.com/Homebrew/brew/runs/8023876296?check_suite_focus=true#step:5:8
After merging PR https://github.com/Homebrew/brew/pull/13577 `install glibc/gcc automatically if too old` this PR now fails with  `Error: maximum tree depth (10) exceeded calculating gcc@11 dependency tree!`.

**Comment by sjackman** [comment]
### Bump `LINUX_GLIBC_CI_VERSION`
```diff
-  LINUX_GLIBC_CI_VERSION = "2.23"
+  LINUX_GLIBC_CI_VERSION = "2.35"
```
```console
$ brew install hello
Error: maximum tree depth (10) exceeded calculating gcc@5 dependency tree!
```

### Bump `LINUX_GCC_CI_VERSION`
```diff
-  LINUX_GCC_CI_VERSION = "5.0"
+  LINUX_GCC_CI_VERSION = "11.0"
```

```console
$ brew install hello
…
==> Installing dependencies for hello: gmp, isl@0.18, mpfr, libmpc, zlib, binutils, linux-headers@5.15, glibc and gcc@5
```

**Comment by MikeMcQuaid** [comment]
> After merging PR #13577 this PR now fails with `Error: maximum tree depth (10) exceeded calculating gcc@11 dependency tree!`.

@sjackman Going to need you to dig in a bit harder on these sorts of issues, please. I don't even really have a Linux machine to test these on. If the value for that variable needs changed: consider changing it as part of this PR.

**Comment by sjackman** [comment]
> Going to need you to dig in a bit harder on these sorts of issues, please. I don't even really have a Linux machine to test these on. If the value for that variable needs changed: consider changing it as part of this PR.

https://docs.docker.com/desktop/install/mac-install/
I do all of my testing on Linux using Docker Desktop for macOS and `docker run -it --rm homebrew/ubuntu16.04`.

The circular dependencies in Homebrew/core were resolved by (I believe)…
- https://github.com/Homebrew/homebrew-core/pull/109173
- https://github.com/Homebrew/homebrew-core/pull/109186
- https://github.com/Homebrew/homebrew-core/pull/109189

Thank you @Bo98 and @danielnachun!

File: .github/workflows/tests.yml
Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:189]: @sjackman I consider this unresolved comment to be blocking the 3.6.0 release, let's figure out a backwards compatibility/deprecation story here please.

**Comment by MikeMcQuaid** [comment]
> > Going to need you to dig in a bit harder on these sorts of issues, please. I don't even really have a Linux machine to test these on. If the value for that variable needs changed: consider changing it as part of this PR.
> 
> https://docs.docker.com/desktop/install/mac-install/ I do all of my testing on Linux using Docker Desktop for macOS and `docker run -it --rm homebrew/ubuntu16.04`.

@sjackman This was not a request for you to tell me how to obtain a Linux machine (which runs incredibly slowly/buggily on all my ARM macOS machines) but instead a request for you as a maintainer to be able to attempt to resolve issues like that you reported yourself if they are blocking your PRs in future, thanks ❤️

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
Blocked on:
- figuring out what to do with `homebrew/ubuntu16.04:master` vs. `homebrew/ubuntu16.04:latest`
- legitimate `brew tests` failures

File: .github/workflows/tests.yml
Comment (Bo98) [reply .github/workflows/tests.yml:189]: If there's a desire to deprecate (not clear whether that's just `master` or `ubuntu16.04` as a whole), we could implemented a system in brew where we can insert a gitignored file into the Docker image that will add a deprecation message to each invokation of `brew` (and as a login shell message too if you'd like).

This can be tested out with `Homebrew/debian7`, which we don't need to provide a deprecation notice for but we can use it as a test.

**Comment by Bo98** [comment]
That's quite a strong argument to me to use Ubuntu 20.04 as a base. I've also realised earlier today that Homebrew is now likely going to be inoperable on devices like Raspberry Pi because of the Glibc changes (there's no stable Debian version with Glibc 2.35 and our new Glibc formula looks like it'll only work on x86_64).

**Comment by danielnachun** [comment]
> That's quite a strong argument to me to use Ubuntu 20.04 as a base. I've also realised earlier today that Homebrew is now likely going to be inoperable on devices like Raspberry Pi because of the Glibc changes (there's no stable Debian version with Glibc 2.35 and our new Glibc formula looks like it'll only work on x86_64 unlike before).

We could just condition the installation of the bootstrap binaries in `glibc` on the host CPU being `x86_64`, and it would work fine on ARM, assuming the user has the necessary dependencies.

If we do go with 20.04 I do want to note that we already have a few formulae that need GCC 10+ for C++20 support.  I think we should have an allowlist for them to use brewed GCC if we only have GCC 9 available in CI.  It's a small number of formulae that should pose no issue for us to manage in CI, certainly nothing like it is right now, and it would go away once we upgrade.

Perhaps the strategy we should use is that our CI image is the same version as whatever `ubuntu-latest` is for GitHub Actions.  Assuming the same schedule is followed as with 20.04, I would anticipate them switching to 22.04 at the end of the year.  That falls well within our window of one year after the release of the latest LTS.  I would hope by then that Debian had gotten out a version with `glibc` 2.35 or newer as well, and I would expect most other major Linux distros will have a stable release with a new enough GCC.

I actually think it might be a decent idea to upgrade the `glibc` and `linux-headers` formulae to the version that will be used in the newest LTS before migrating to that version. That gives us a chance to test how things will work and deal with deprecations etc.  Then when the time comes to actually migrate, we only have to bump the relevant versions in `brew` and `homebrew-core`.  We've done an incredible job cleaning up technical debt already so I do not anticipate the migration being anywhere near as difficult as it was this time around.

File: .github/workflows/tests.yml
Comment (danielnachun) [reply .github/workflows/tests.yml:189]: I think we should issue a deprecation notice for `ubuntu16.04`, and one for `ubuntu18.04` when it goes EOL next year as well.  Testing on `debian7` makes sense, because we no longer use it for anything.

**Comment by MikeMcQuaid** [comment]
> The GitHub Actions VM image includes Homebrew. Users of GitHub Actions who are using runs-on: ubuntu-latest and brew install will first download glibc and gcc before install the desired package, which substantially increases the time needed to install the desired package. 

There's a trivial workaround for this which will happen quickly: `glibc` and `gcc` are installed in the Homebrew Ubuntu image by default. We should not delay or change this migration due to a short-term disruption like this.

> I've also realised earlier today that Homebrew is now likely going to be inoperable on devices like Raspberry Pi because of the Glibc changes (there's no stable Debian version with Glibc 2.35 and our new Glibc formula looks like it'll only work on x86_64 unlike before).

We already attempt to support too much. Until we have Linux ARM CI: we should not be blocking upgrades based on unsupported and untested platforms.

> We could just condition the installation of the bootstrap binaries in glibc on the host CPU being x86_64, and it would work fine on ARM, assuming the user has the necessary dependencies.

This is fine with me as long as it happens after the migration, not before. It's really not a good use to time today for, again, an unsupported platform.

> Perhaps the strategy we should use is that our CI image is the same version as whatever ubuntu-latest is for GitHub Actions. 

Definitely not. It makes zero sense to me to gate our upgrades on GitHub's.

> I actually think it might be a decent idea to upgrade the glibc and linux-headers formulae to the version that will be used in the newest LTS before migrating to that version. 

@danielnachun is this not what we've done/are doing?

**Comment by Bo98** [comment]
> > 
> 
> We already attempt to support too much. Until we have Linux ARM CI: we should not be blocking upgrades based on unsupported and untested platforms.

Just to clarify here, given there's multiple definitions of "unsupported": ARM on Linux has the same wording as macOS 10.11-10.14. https://docs.brew.sh/Homebrew-on-Linux#arm

> We could just condition the installation of the bootstrap binaries in glibc on the host CPU being x86_64, and it would work fine on ARM, assuming the user has the necessary dependencies.

Fine with me.

**Comment by sjackman** [comment]
>> The GitHub Actions VM image includes Homebrew. Users of GitHub Actions who are using runs-on: ubuntu-latest and brew install will first download glibc and gcc before install the desired package, which substantially increases the time needed to install the desired package.

> There's a trivial workaround for this which will happen quickly: `glibc` and `gcc` are installed in the Homebrew Ubuntu image by default. We should not delay or change this migration due to a short-term disruption like this.

Installing `glibc` and `gcc` in the `homebrew/ubuntu20.04` image does not directly help users of the GitHub Actions VM image `ubuntu-latest` without a change to those users' CI configuration. GitHub would need to install `glibc` and `gcc` in the GitHub Actions VM image. With a configuration change, users can switch from `runs-on: ubuntu-latest` to `runs-on: ubuntu-22.04`.

> The workaround is for GitHub Actions users currently using `runs-on: ubuntu-latest` to switch to `runs-on: ubuntu-22.04`. The workaround is nice and simple, but it's possible that this workaround may not be an option for some users for reasons related to other tools outside of Homebrew.

I think you may have been suggesting this: Installing `glibc` and `gcc` in the `homebrew/ubuntu20.04` image does provide another workaround for users who are not able to update to `runs-on: ubuntu-22.04` for whatever reason. Those users can instead switch from using `runs-on: ubuntu-latest` to using `image: ghcr.io/homebrew/ubuntu20.04`.

**Comment by MikeMcQuaid** [comment]
> Installing `glibc` and `gcc` in the `homebrew/ubuntu20.04` image does not directly help users of the GitHub Actions VM image `ubuntu-latest` without a change to those users' CI configuration. GitHub would need to install `glibc` and `gcc` in the GitHub Actions VM image. With a configuration change, users can switch from `runs-on: ubuntu-latest` to `runs-on: ubuntu-22.04`.

Sorry, yes, this is exactly what I was saying will happen quickly from GitHub.

> I think you may have been suggesting this

I wasn't but this is a good idea, too.

**Comment by danielnachun** [comment]
I have no strong feelings about `glibc` and `gcc` needing to be installed in CI workflows.  I think it probably does make sense for our older CI images to have `glibc` and `gcc` preinstalled but we don't have to delay the migration to handle that.

I'll open a PR for `glibc` to only bootstrap on Intel.  It won't require any new bottles so we can merge it right it away.

>> I actually think it might be a decent idea to upgrade the glibc and linux-headers formulae to the version that will be used in the newest LTS before migrating to that version.
> @danielnachun is this not what we've done/are doing?

I don't think we've officially said we are going to do it this way in the future but we should.  Basically there's no reason not to upgrade `linux-headers` and `glibc` to the versions we know we will need for the latest Ubuntu LTS once the versions have been announced.  That lets us separate out the `glibc`/`linux-headers` upgrade step from the actual migration step instead of it all happening at the same time which is more chaotic.

On a related note, I think we should also make it an official policy that if `glibc` deprecates one its components, we immediately initiate a deprecation period of specified length of time where a warning is issued, after which the warning becomes a failure.  That should reduce the likelihood of us having to deal with a `glibc` component deprecation at the same time as a migration.  The `libcrypt` and `libnsl` deprecations happened long ago which resulted in us accumulating technical debt to get rid of them and we should avoid letting this happen in the future by being proactive about it.

**Comment by sjackman** [comment]
> Blocked on: figuring out what to do with `homebrew/ubuntu16.04:master` vs. `homebrew/ubuntu16.04:latest`

Here are five options in my order of preference:
Option 1: Replace the image with `echo "homebrew/ubuntu16.04:master is retired. Use homebrew/brew or homebrew/ubuntu16.04"; exit 1`
Option 2: Delete `homebrew/ubuntu16.04:master`
Option 3: Continue updating it with `homebrew/ubuntu16.04:latest`
Option 4: Continue updating it with a build of the `master` branch
Option 5: Keep it and stop updating it

**Comment by carlocab** [comment]
> Here are five options in my order of preference:

Happy with option 1 as well. Could you clarify what happens with the `latest` tag under this option?

File: .github/workflows/tests.yml
Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:189]: I wonder whether we even need to deprecate these rather than (as @sjackman has proposed below and I'd be favour of) just make them throw an error message with `master` or `latest`.

**Review by MikeMcQuaid — COMMENTED** [review]
Fine with me once we have the Ubuntu deprecation/error message handled.

**Comment by MikeMcQuaid** [comment]
> > Here are five options in my order of preference:
> 
> Happy with option 1 as well. Could you clarify what happens with the `latest` tag under this option?

Also happy with option 1 👍🏻

**Comment by sjackman** [comment]
> Happy with option 1 as well. Could you clarify what happens with the `latest` tag under this option?

No change to `latest`. CI pushes a `latest` image for Ubuntu LTS versions 16.04, 18.04, 20.04, and 22.04.

> Fine with me once we have the Ubuntu deprecation/error message handled.

I can push that error message image shown above to Docker Hub and GHCR once this PR has been merged and `master` CI has built and pushed the new `homebrew/ubuntu22.04:master` image.

**Comment by Bo98** [comment]
I'm fine with aiming for option 1, as long as whatever we do will show the message when run in GitHub Actions too.

Though I do feel there should be a crossover deprecation period where everything still works but the message is also displayed. Which would basically be a hybrid of option 1 and either option 3 or 4 initially. We have deprecation notice periods for everything else we do, and I feel the Docker image should be no different.

This deprecation period could be a month or so and then we push a final update to the image that disables it entirely.

**Comment by danielnachun** [comment]
This is probably a question for later, but for how much longer do we plan to continue building an image for 16.04 at all, given that it is EOL?

**Comment by sjackman** [comment]
> This is probably a question for later, but for how much longer do we plan to continue building an image for 16.04 at all, given that it is EOL?

https://ubuntu.com/about/release-cycle
Ubuntu 16.04 is not EOL. It's Extended Security Maintenance (ESM) until 2026.
image

**Comment by sjackman** [comment]
> Though I do feel there should be a crossover deprecation period where everything still works but the message is also displayed.

I'm not against this suggestion, but I really doubt that anyone will see the message so long as their CI job is still succeeding.

**Comment by sjackman** [comment]
A couple of minor bug reports:
```console
$ ls -d /mnt/home/shaun.jackman/.linuxbrew/Cellar/linux-headers*
/mnt/home/shaun.jackman/.linuxbrew/Cellar/linux-headers@4.4
/mnt/home/shaun.jackman/.linuxbrew/Cellar/linux-headers@5.15
$ readlink /mnt/home/shaun.jackman/.linuxbrew/Cellar/linux-headers
linux-headers@4.4
```
This symlink `linux-headers` in `Cellar` was not updated to point to `linux-headers@5.15`. I don't know what process created this symlink, or what process is meant to update it.

```console
$ ls ~/.linuxbrew/lib/libstdc++.so.6
ls: cannot access '/home/shaun.jackman/.linuxbrew/lib/libstdc++.so.6': No such file or directory
```
This symlink `$HOMEBREW_PREFIX/lib/libstdc++.so.6` was not created as expected.

**Comment by Bo98** [comment]
> > This is probably a question for later, but for how much longer do we plan to continue building an image for 16.04 at all, given that it is EOL?
> 
> https://ubuntu.com/about/release-cycle Ubuntu 16.04 is not EOL. It's Extended Security Maintenance (ESM) until 2026. Still a valid question though.

Does ESM apply to the public Docker images?

> > Though I do feel there should be a crossover deprecation period where everything still works but the message is also displayed.
> 
> I'm not against this suggestion, but I doubt that anyone will see the message so long as their CI job is still succeeding.

We could do something clever here like with GitHub annotations. I will experiment with ideas today and report back.

> This symlink `linux-headers` in `Cellar` was not updated to point to `linux-headers@5.15`. I don't know what process created this symlink, or what process is meant to update it.

Should be controlled by `Aliases/linux-headers`. Not sure why it's not updated though.

**Comment by sjackman** [comment]
> https://ubuntu.com/about/release-cycle Ubuntu 16.04 is not EOL. It's Extended Security Maintenance (ESM) until 2026. Still a valid question though.
> Does ESM apply to the public Docker images?

https://hub.docker.com/_/ubuntu
The web page says that everything from Ubuntu 14.04 LTS and up is supported.

**Comment by Bo98** [comment]
> > > https://ubuntu.com/about/release-cycle Ubuntu 16.04 is not EOL. It's Extended Security Maintenance (ESM) until 2026. Still a valid question though.
> 
> > Does ESM apply to the public Docker images?
> 
> https://hub.docker.com/_/ubuntu The web page says that everything from Ubuntu 14.04 LTS and up is currently supported.

The datestamps in the tags there are 2019 and 2021 and ESM usually requires login and attaching a device with a token (`ua attach ...`) so was wondering how that exactly worked, particularly with the device limit.

Seems quite vague. Their page on Docker images suggests 10 year support requires contacting them: https://ubuntu.com/security/docker-images

**Comment by sjackman** [comment]
I believe this PR is ready to be merged and Homebrew 3.6.0 tagged!

**Comment by MikeMcQuaid** [comment]
> I can push that error message image shown above to Docker Hub and GHCR once this PR has been merged and `master` CI has built and pushed the new `homebrew/ubuntu22.04:master` image.

@sjackman Could this be made part of this PR instead? If not, presumably this needs to happen before 3.6.0 is merged rather than after? Ideally I think it'd be done in a PR merged before this one.

**Comment by sjackman** [comment]
>> I can push that error message image shown above to Docker Hub and GHCR once this PR has been merged and `master` CI has built and pushed the new `homebrew/ubuntu22.04:master` image.

> @sjackman Could this be made part of this PR instead? If not, presumably this needs to happen before 3.6.0 is merged rather than after? Ideally I think it'd be done in a PR merged before this one.

https://github.com/Homebrew/brew/pull/13733#issuecomment-1234709417
I was planning on pushing this retired-message Docker image from the interactive command line using `docker login && docker push homebrew/ubuntu16.04:master` after we'd successfully built the Docker image `homebrew/ubuntu22.04:master` by merging this PR, and merged PR https://github.com/Homebrew/homebrew-core/pull/108590, and successfully used it to build at least one bottle.
- https://github.com/Homebrew/homebrew-core/pull/108590

### Tasks

- [ ] Merge this PR https://github.com/Homebrew/brew/pull/13733
- [ ] Rerun and merge PR https://github.com/Homebrew/homebrew-core/pull/108590
- [ ] Build at least one bottle successfully using that image (bottle CI is 💚)
- [ ] Tag Homebrew 3.6.0
- [ ] Merge that bottle PR and test it
- [ ] Push the retired-message Docker image to `homebrew/ubuntu16.04:master`

**Review by MikeMcQuaid — APPROVED** [review]
> I was planning on pushing this retired-message Docker image from the interactive command line using `docker login && docker push homebrew/ubuntu16.04:master` after we'd successfully built the Docker image `homebrew/ubuntu22.04:master` by merging this PR, and merged PR [Homebrew/homebrew-core#108590](https://github.com/Homebrew/homebrew-core/pull/108590), and successfully used it to build at least one bottle.

@sjackman Could this be done by a/the GitHub Action in a new PR instead of by you manually through the CLI? I think that'd be preferable.

This does not need to block this PR but would be good to be open before the tag.

---

FYI that we've landed some fairly major other PRs today that I want to be merged for ~24h before we tag 3.6.0 to surface any issues.

**Comment by MikeMcQuaid** [comment]
> This branch has conflicts that must be resolved

Note that these need to be addressed, sorry!



> FYI that we've landed some fairly major other PRs today that I want to be merged for ~24h before we tag 3.6.0 to surface any issues.

Sorry, the specific question here: do we need to tag ASAP after merging this and/or https://github.com/Homebrew/homebrew-core/pull/108590?

**Comment by sjackman** [comment]
> Sorry, the specific question here: do we need to tag ASAP after merging this and/or https://github.com/Homebrew/homebrew-core/pull/108590?

After merging PR https://github.com/Homebrew/homebrew-core/pull/108590, we need to tag Homebrew 3.6.0 ASAP before merging any bottle PRs in Homebrew/core.
- https://github.com/Homebrew/homebrew-core/pull/108590

**Comment by MikeMcQuaid** [comment]
> > Sorry, the specific question here: do we need to tag ASAP after merging this and/or [Homebrew/homebrew-core#108590](https://github.com/Homebrew/homebrew-core/pull/108590)?
> 
> After merging PR [Homebrew/homebrew-core#108590](https://github.com/Homebrew/homebrew-core/pull/108590), we need to tag Homebrew 3.6.0 ASAP before merging any bottle PRs in Homebrew/core.

Thanks. What about merging this PR? Could it be done 1/12/24 hours before https://github.com/Homebrew/homebrew-core/pull/108590?

**Comment by carlocab** [comment]
> > Sorry, the specific question here: do we need to tag ASAP after merging this and/or [Homebrew/homebrew-core#108590](https://github.com/Homebrew/homebrew-core/pull/108590)?
> 
> After merging PR [Homebrew/homebrew-core#108590](https://github.com/Homebrew/homebrew-core/pull/108590), we need to tag Homebrew 3.6.0 ASAP before merging any bottle PRs in Homebrew/core.
> 
> * [Use Ubuntu 22.04 to build bottles homebrew-core#108590](https://github.com/Homebrew/homebrew-core/pull/108590)

It seems like we should tag 3.6.0 before merging Homebrew/homebrew-core#108590, but the merge should be done quickly after the tag in order to avoid having to install `glibc` and `gcc@11` in CI.

File: Library/Homebrew/test/unpack_strategy/zstd_spec.rb
Comment (sjackman) [reply Library/Homebrew/test/unpack_strategy/zstd_spec.rb:11]: `UnpackStrategy.detect(path)` for a `.tar.XXX` file returns either `UnpackStrategy::Tar` if the host's `tar` is able to extract that compressed file or `UnpackStrategy::XXX` otherwise (such as `UnpackStrategy::Zstd`). On macOS `UnpackStrategy.detect("container.tar.zst")` returns `UnpackStrategy::Zstd`, and on `ubuntu-22.04` it returns `UnpackStrategy::Tar`, because the host's version of `tar` is recent enough and `zstd` is installed.

Comment (carlocab) [reply Library/Homebrew/test/unpack_strategy/zstd_spec.rb:11]: This seems worth documenting in a comment, if not adjusted in `shared_examples.rb`.

Comment (sjackman) [reply Library/Homebrew/test/unpack_strategy/zstd_spec.rb]: Done!

**Comment by sjackman** [comment]
```console
$ brew tests --only=cask/audit
Failed examples:
[160](https://github.com/Homebrew/brew/runs/8196408355?check_suite_focus=true#step:14:161)

[161](https://github.com/Homebrew/brew/runs/8196408355?check_suite_focus=true#step:14:162)
rspec ./test/cask/audit_spec.rb[1:3:1:2:1] # Cask::Audit#run! required stanzas when missing sha256 is expected to fail with /sha256 stanza is required/
[162](https://github.com/Homebrew/brew/runs/8196408355?check_suite_focus=true#step:14:163)
rspec ./test/cask/audit_spec.rb[1:3:1:4:1] # Cask::Audit#run! required stanzas when missing name is expected to fail with /name stanza is required/
[163](https://github.com/Homebrew/brew/runs/8196408355?check_suite_focus=true#step:14:164)
rspec ./test/cask/audit_spec.rb[1:3:1:5:1] # Cask::Audit#run! required stanzas when missing homepage is expected to fail with /homepage stanza is required/
[164](https://github.com/Homebrew/brew/runs/8196408355?check_suite_focus=true#step:14:165)
rspec ./test/cask/audit_spec.rb[1:3:1:3:1] # Cask::Audit#run! required stanzas when missing url is expected to fail with /url stanza is required/
[165](https://github.com/Homebrew/brew/runs/8196408355?check_suite_focus=true#step:14:166)
rspec ./test/cask/audit_spec.rb:1122 # Cask::Audit#run! checking verified when there is no homepage is expected to fail with /a homepage stanza is required/
[166](https://github.com/Homebrew/brew/runs/8196408355?check_suite_focus=true#step:14:167)
```
https://github.com/Homebrew/brew/runs/8196408355?check_suite_focus=true#step:14:160

I"m going to need some help troubleshooting these tests that are failing now.

**Comment by carlocab** [comment]
Failures likely fixed by #13813.

**Pull request merged**
